### DISCUSSION TITLE
pipe_execute(cmd, []byte) & stream interface & pipe_iter_execute(cmd, []byte Iterable)

### DISCUSSION THREAD

**Discussion by gcxfd** [vlang/v#14633] [Ideas]
format v document code is too difficult, it seems that the code blank in the v document with tab, but I vim will always automatically tab to space, which will lead to cmd/tools/vcheck-md.v error

I think this step can be completely automated, the program format markdown and then save, or even add to ci, do not let people to repeat this uninteresting work.

I tried to write a vsh script, but it was blocked at this step and I couldn't get execute to receive stdin pipe input.

I see that the current implementation of v check-md is to extract the v code in the document to a temporary file save to disk, and then call `v fmt`

I don't think it's necessary to write file to disk at all, it's inelegant

`v fmt` is available `cat test.v|v fmt`.

However, there is no pipe_execute(cmd,string) function.

Can someone please add a function like this? I've read the code and it doesn't clear how it should be implemented.

Also, does the v language now have a concept like [stream](https://nodejs.org/api/stream.html#stream) in nodejs?

For example, if I want to count the number of lines based on pipe for a 200GB file, I think pipe_iter_execute(cmd, []byte Iterable) is more appropriate.

**Comment by dumblob** [comment]
Please first split this issue into several separate ones to make it easier to tackle each request. Thank you!

**Comment by ken0x0a** [comment]
@gcxfd 
Is this https://github.com/vlang/v/pull/8432 solve your goal?
